WordPress/Classes/Extensions/WKWebView+SynchJSEvaluation.swift

```diff
@@ -0,0 +1,47 @@
+import AutomatticTracks
+import WebKit
+
+/// This extension contains a hack used to make it possible to
+/// evaluate JavaScript code synchronously.
+///
+extension WKWebView {
+
+    /// Synchornous evaluation of JavaScript in WKWebView.
+    ///
+    /// - Parameters:
+    ///     - javascriptString: the JavaScript string to evaluate.
+    ///
+    /// - Returns: the result of the evaluation.
+    ///
+    @objc
+    func stringByEvaluatingJavaScript(fromString javascriptString: String) -> String {
+        var result = ""
+        var finished = false
+
+        evaluateJavaScript(javascriptString) { (output, error) in
+            defer {
+                finished = true
+            }
+
+            guard error == nil else {
+                CrashLogging.logMessage(
```

Comment:
Just out of curiosity: we're logging here as a crash. But this will not cause a crash, will it?

Reply:
We're not exactly logging here as a "crash" but as an error.  I believe Sentry should show this as a non crashing error.

That said the reason why logging this is important is because we don't want this to fail silently if there are issues evaluating the Javascript.  This could happen eventually due to typos in the Javascript string, or due to things like changing APIs.